pkg/local_object_storage/shard/gc.go

```diff
@@ -171,7 +171,12 @@ func (gc *gc) stop() {
 
 // iterates over metabase graveyard and deletes objects
 // with GC-marked graves.
+// Does nothing if shard is in "read-only" mode.
 func (s *Shard) removeGarbage() {
+	if s.getMode() == ModeReadOnly {
+		return
```

Comment:
Shouldn't we panic here?

Reply:
This PR doesn't stop GC worker, thus it is a valid case to call `removeGarbage` on read-only shard. We can't easily stop and start GC worker, it will require a bit of change in internal components (remove some `sync.Once` code), so we can postpone it for separate PR.